test/cmocka/cmocka-host-gcc-toolchain.cmake

```diff
@@ -0,0 +1,28 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+message(STATUS "Preparing Host GCC toolchain")
+
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_SYSTEM_VERSION 1)
+
+set(CMAKE_ASM_COMPILER_FORCED 1)
+set(CMAKE_C_COMPILER_FORCED 1)
+
+set(CMAKE_ASM_COMPILER_ID GNU)
+set(CMAKE_C_COMPILER_ID GNU)
+
+set(CMAKE_C_COMPILER gcc)
+
+find_program(CMAKE_LD NAMES "$ld" PATHS ENV PATH NO_DEFAULT_PATH)
+find_program(CMAKE_AR NAMES "$ar" PATHS ENV PATH NO_DEFAULT_PATH)
+find_program(CMAKE_OBJCOPY NAMES "objcopy" PATHS ENV PATH NO_DEFAULT_PATH)
+find_program(CMAKE_OBJDUMP NAMES "objdump" PATHS ENV PATH NO_DEFAULT_PATH)
```

Comment:
this might well be correct, just a bit strange why we use `NO_DEFAULT_PATH`?

Reply:
This aligns with the current code for XCC and can be used to use to specify specific CC versions (which is already supported as one of the cmocka cmake options). We can always revisits if this is a problem later, but it seems OK atm.